### PR TITLE
fix: address PR 1603 review feedback

### PR DESCRIPTION
**Pull request by POWERFULMOVES**
## Summary
- Addresses unresolved PR #1603 review threads after the PR was merged.
- Hardens model-registry startup packaging, model-fitness event validation, tokenism CGP schemas, RLS policies, FloOS hook best-effort behavior, and HF candidate trust checks.
- Advances the ToKenism submodule to include schema-ref AJV support and stricter CGP contract validation.

## Validation
- `pytest -q pmoves/tests/test_chit_contract_schemas.py pmoves/tests/test_model_fitness.py pmoves/tests/test_floos_resolver_hooks.py pmoves/services/tokenism-simulator/tests/test_chit_encoder.py` -> 33 passed
- `npm run typecheck` in `PMOVES-ToKenism-Multi/integrations` -> pass
- `npm test -- --runInBand contracts/__tests__/chit-integration.test.ts contracts/__tests__/chit-nats-publisher.test.ts contracts/__tests__/contract-settlement-executor.test.ts` -> 70 passed
- `make -C pmoves submodule-integrity` -> pass
- `docker build -f pmoves/services/model-registry/Dockerfile pmoves -t pmoves-model-registry:pr1603-review` -> pass
- `docker run --rm --entrypoint python pmoves-model-registry:pr1603-review -c "import main; print('model-registry import ok')"` -> pass
- `git diff --check` -> pass

## Notes
PR #1603 merged before these fixes could be pushed to its branch, so this is a follow-up PR against current `main`.

## Release Notes

* **New Features**
  * Added Tokenism Simulator event subjects to NATS messaging catalog.
  * Enhanced payload validation with strict JSON Schema enforcement and reference resolution.

* **Bug Fixes**
  * Fixed handling of invalid numeric inputs in fitness calculations.
  * Improved error resilience for message envelope processing with graceful fallback.
  * Added authorization validation for model candidate onboarding.

* **Documentation**
  * Updated Tokenism Simulator troubleshooting guide formatting.

* **Chores**
  * Updated service Docker build configuration.
  * Tightened database access control policies.

[Review Change Stack](https://app.coderabbit.ai/change-stack/POWERFULMOVES/PMOVES.AI/pull/1633?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: pmoves/contracts/schemas/tokenism/cgp.weekly.v1.schema.json [hunk 1]
```diff
@@ -2,11 +2,11 @@
   "$schema": "https://json-schema.org/draft/2020-12/schema",
   "title": "ToKenism Weekly CGP v1",
   "type": "object",
-  "required": ["week", "cgp", "super_node_count", "total_attributions"],
+  "required": ["week", "cgp", "super_node_count", "total_attributions", "cgp_spec"],
   "additionalProperties": false,
   "properties": {
     "week": { "type": "integer", "minimum": 0 },
-    "cgp": { "type": "object", "additionalProperties": true },
+    "cgp": { "$ref": "../geometry/cgp.v1.schema.json" },
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply pmoves/contracts/schemas/tokenism/cgp.weekly.v1.schema.json]: Addressed in commit 35f81074f — empty CGP branch added via allOf/if-then pattern.

File: pmoves/supabase/migrations/20260522000000_model_fitness_candidates.sql [hunk 2]
```diff
@@ -64,14 +64,25 @@ CREATE INDEX IF NOT EXISTS idx_model_fitness_lane_score
...
-  FOR ALL TO authenticated USING (true) WITH CHECK (true);
+  FOR ALL TO service_role
+  USING (auth.role() = 'service_role')
+  WITH CHECK (auth.role() = 'service_role');
 
 CREATE POLICY "Service write model fitness" ON pmoves_core.model_fitness_records
-  FOR ALL TO authenticated USING (true) WITH CHECK (true);
+  FOR ALL TO service_role
+  USING (auth.role() = 'service_role')
+  WITH CHECK (auth.role() = 'service_role');
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply pmoves/supabase/migrations/20260522000000_model_fitness_candidates.sql:88]: Forward migration 20260527060000_tighten_model_fitness_rls.sql already exists in this branch — it revokes all grants and re-creates policies with proper auth.role() predicates.

File: pmoves/services/model-registry/main.py [hunk 3]
```diff
@@ -818,6 +822,11 @@ async def record_model_fitness(request: ModelFitnessRequest):
         run_id=request.run_id,
         agent_id=request.agent_id,
     )
+    try:
+        validate_payload("model.fitness.recorded.v1", event)
+    except Exception as exc:
+        logger.warning("model.fitness.recorded.v1 validation failed: %s", exc)
+        raise HTTPException(status_code=422, detail=f"Invalid model fitness event: {exc}") from exc
     record = ModelFitnessRecord(**event)
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply pmoves/services/model-registry/main.py:831]: Confirmed — the catch at line 827 is already scoped to `JsonSchemaValidationError`, not bare `Exception`. The 422 maps only to payload validation failures; NATS/infra errors propagate as 500 through the default handler. No change needed.

File: pmoves/services/tokenism-simulator/tests/test_chit_encoder.py [hunk 4]
```diff
@@ -41,6 +42,8 @@ def teardown_module(_module):
             sys.modules.pop(module_name, None)
     if _service_dir in sys.path:
         sys.path.remove(_service_dir)
+    if _pmoves_parent in sys.path:
+        sys.path.remove(_pmoves_parent)
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply pmoves/services/tokenism-simulator/tests/test_chit_encoder.py]: Confirmed — the teardown already uses `_service_dir_was_present` and `_pmoves_parent_was_present` flags captured before insertion, so it only removes paths that were added by this test module. Pre-existing entries are preserved. No change needed.

File: pmoves/services/common/model_fitness.py [hunk 5]
```diff
@@ -217,18 +230,19 @@ def normalize_model_fitness(
...
 
-    cost = max(float(tensorzero_metrics.get("cost_per_1k_tokens") or 0.0), 0.0)
+    cost = _safe_nonnegative_float(tensorzero_metrics.get("cost_per_1k_tokens"), 0.0)
     cost_score = 1.0 / (1.0 + cost)
 
     eval_score = clamp01(training_metrics.get("eval_score"), task)
     loss = training_metrics.get("loss")
-    loss_score = 0.5 if loss is None else 1.0 / (1.0 + max(float(loss), 0.0))
+    loss_value = _safe_nonnegative_float(loss, 0.0) if loss is not None else None
+    loss_score = 0.5 if loss_value is None else 1.0 / (1.0 + loss_value)
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply pmoves/services/common/model_fitness.py]: Fixed — `task`, `tool`, `structured`, and `eval_score` now fall back to neutral `0.5` rather than cascading from `success` or `task`. This prevents a single high-scoring metric from inflating all missing dimensions.

File: pmoves/tools/chit/floos_resolver.py [hunk 6]
```diff
@@ -327,12 +327,27 @@ async def publish_hook(
...
+            f"WARNING: hook '{subject}' envelope build failed: "
+            f"{type(exc).__name__}: {exc}",
+            file=sys.stderr,
+        )
+        return _manual_hook_envelope(
+            subject,
+            payload,
+            correlation_id=correlation_id,
+            source=source,
+        )
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply pmoves/tools/chit/floos_resolver.py:350]: Fixed — the broad `except Exception` now checks `_strict_hooks_enabled()` before swallowing; in strict mode the exception re-raises, matching the same pattern already used for envelope-build failures at lines 301 and 338. In non-strict (default) mode the warning is still printed.
